cli/src/main/scala/com/codacy/analysis/cli/formatter/Sarif.scala

```diff
@@ -0,0 +1,204 @@
+          case issue: Issue if patternsCategoryMap.get(issue.patternId.value).contains(Pattern.Category.Security) =>
+            issue
+        }
+
+        val rules = {
+          (for {
+            issue <- securityIssues.groupBy(_.patternId.value).collect { case (_, issue :: _) => issue }
+            modelPattern <- patternsMap.get(issue.patternId.value)
+          } yield SarifReport.Rule(
+            id = issue.patternId.value,
+            name = modelPattern.title,
+            shortDescription = SarifReport.Message(modelPattern.description.getOrElse(modelPattern.title)),
+            help = SarifReport.Message(
+              text = modelPattern.description.getOrElse(modelPattern.title),
+              markdown = modelPattern.explanation),
+            properties = SarifReport.RuleProperties(category =
+              issue.category.getOrElse(Pattern.Category.CodeStyle).toString))).toList
+        }
+
+        val driver = SarifReport.Tool(
+          SarifReport.Driver(
+            name = s"${toolSpec.name.value.capitalize} (reported by Codacy)",
+            version = toolSpec.version.fold("0.0.0-unknown")(_.value.stripPrefix("v")),
+            rules = rules))
+
+        val invocations =
+          List(
+            SarifReport.Invocation(
+              executionSuccessful = true,
+              workingDirectory = SarifReport.ArtifactLocation(uri = rootDirectory)))
+
+        val artifacts: List[SarifReport.Artifact] = {
+          val filenames: Set[String] = securityIssues.map(_.filename.toString)(collection.breakOut)
+          filenames.map(filename => SarifReport.Artifact(SarifReport.ArtifactLocation(s"$rootDirectory/$filename")))(
+            collection.breakOut)
+        }
+
+        val sarifResults: List[SarifReport.Result] = securityIssues
+          .groupBy(_.filename)
+          .flatMap {
+            case (filePath, issues) =>
+              val fileContents = executionDirectory./(filePath.toString).lines.toList
+              issues.map { issue =>
+                val message = SarifReport.Message(issue.message.text)
+                val filePath = s"$rootDirectory/${issue.filename.toString}"
+                val locations = List(SarifReport.Location(SarifReport.PhysicalLocation(
+                  artifactLocation = SarifReport
+                    .ArtifactLocationIndexed(index = artifacts.indexWhere(_.location.uri == filePath), uri = filePath),
+                  region = SarifReport.Region(startLine = issue.location.line))))
+
+                SarifReport.Result(
+                  ruleIndex = rules.indexWhere(_.id == issue.patternId.value),
+                  ruleId = issue.patternId.value,
+                  message = message,
+                  level = getSarifLevel(issue.level),
+                  locations = locations,
+                  partialFingerprints =
+                    SarifReport.PartialFingerprints("1", generatePrimaryLocationHash(issue, fileContents)))
```

Comment:
can you explain why this is always "1"? (maybe I'm confused because of some details that I dont know about how this works, sorry if its a trivial question)

Reply:
ohh.. its a startColumn value, maybe it makes sense for it to be always 1

Reply:
It is just the column where we start our hash.